internal/workload/wrapper.go

```diff
@@ -133,7 +134,12 @@ func (ww Workload) Run(workload *v1.Pod, manifestPath string, authFilePath strin
 	if err != nil {
 		return err
 	}
-	return ww.mappingRepository.Add(workload.Name, podIds[0])
+
+	for _, observer := range ww.observers {
+		observer.WorkloadStarted(workload.Name, podIds)
```

Comment:
Will we be able to catch changes to the underlying containers if they were restarted by podman?

Reply:
This is also raised by @jakub-dzon if the connection is not routable it'll fetch again the container IPs.